zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/user/DbUserState.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.state.user;
+
+import static io.camunda.zeebe.util.buffer.BufferUtil.wrapString;
+
+import io.camunda.zeebe.db.ColumnFamily;
+import io.camunda.zeebe.db.TransactionContext;
+import io.camunda.zeebe.db.ZeebeDb;
+import io.camunda.zeebe.db.impl.DbString;
+import io.camunda.zeebe.engine.state.immutable.UserState;
+import io.camunda.zeebe.engine.state.mutable.MutableUserState;
+import io.camunda.zeebe.protocol.ZbColumnFamilies;
+import io.camunda.zeebe.protocol.impl.record.value.user.UserRecord;
+import org.agrona.DirectBuffer;
+
+public class DbUserState implements UserState, MutableUserState {
+
+  private final UserRecordValue userRecordToRead = new UserRecordValue();
+  private final UserRecordValue userRecordToWrite = new UserRecordValue();
```

Comment:
❌ We don't need seperate records for reading and writing. We can define a single `userRecord` and use this for both.

Reply:
Sorry, I wrote this down in a confusing way. Yes I mean we only need to define a single `UserRecordValue` instead of one for reading and one for writing.